### PR TITLE
replacing .format with f-strings

### PR DESCRIPTION
**Pull request by drewseibert**
fixing some styling (using f-strings instead of .format)

### CODE REVIEW

File: pandas/tests/io/test_pickle.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def compare_element(result, expected, typ, version=None):
             assert result.freq == expected.freq
     else:
         comparator = getattr(
-            tm, "assert_{typ}_equal".format(typ=typ), tm.assert_almost_equal
+            tm, f"assert_{typ=typ}_equal", tm.assert_almost_equal
```
Comment (MarcoGorelli) [inline pandas/tests/io/test_pickle.py]: Thanks @drewseibert 

Did you try running this test to see if it works?
```
pytest  pandas/tests/io/test_pickle.py
```

(for more on testing, see [Test-driven development/code writing](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#test-driven-development-code-writing))

Comment (drewseibert) [reply pandas/tests/io/test_pickle.py]: @MarcoGorelli Hi Marco, thanks for taking a look. Just gave it a run locally and I see this:

Traceback (most recent call last):
  File "test_pickle.py", line 25, in <module>
    from pandas.compat import _get_lzma_file, _import_lzma, is_platform_little_endian
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/__init__.py", line 188, in <module>
    v = get_versions()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/_version.py", line 23, in get_versions
    return json.loads(version_json)
AttributeError: module 'json' has no attribute 'loads'

Comment (drewseibert) [reply pandas/tests/io/test_pickle.py]: I don't see any errors related to the f-strings though..

Comment (drewseibert) [reply pandas/tests/io/test_pickle.py]: Sorry, just ran pytest and got this instead...

../../__init__.py:36: in <module>
    f"C extension: {module} not built. If you want to import "
E   ImportError: C extension: No module named 'pandas._libs.tslibs.conversion' not built. If you want to import pandas from the source directory, you may need to run 'python setup.py build_ext --inplace --force' 
to build the C extensions first.

Comment (MarcoGorelli) [reply pandas/tests/io/test_pickle.py]: For C extensions, see [creating a development environment](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#creating-a-development-environment)

Also, you need to run `black` against any Python file you've modified, see [contributing to the code base](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#contributing-to-the-code-base)

Comment (drewseibert) [reply pandas/tests/io/test_pickle.py]: @MarcoGorelli Thanks! Will check those out 👍

**Review by MarcoGorelli — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @drewseibert

**Pull request merged**